Data Prepper documentation updates: autogeneration campaign (#7707)
opensearch-project/documentation-website: _data-prepper/pipelines/configuration/processors/parse_ion.md

<!--
This table is autogenerated. Do not edit it.
- name: parse_ion
- pluginType: processor
- source: https://github.com/opensearch-project/data-prepper/blob/253e59245fd9c39c959c1c8caaeff1b226a5a0ab/data-prepper-plugins/parse-json-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/parse/ion/ParseIonProcessorConfig.java
-->

| Option | Required | Type | Description |
| :--- | :--- | :--- | :--- | 
| `source` | No | String | The field in the `event` that is parsed. Default value is `message`. |
| `destination` | No | String | The destination field of the parsed JSON. Defaults to the root of the `event`. Cannot be `""`, `/`, or any white-space-only `string` because these are not valid `event` fields. |
| `pointer` | No | String | A JSON pointer to the field to be parsed. There is no `pointer` by default, meaning that the entire `source` is parsed. The `pointer` can access JSON array indexes as well. If the JSON pointer is invalid, then the entire `source` data is parsed into the outgoing `event`. If the key that is pointed to already exists in the `event` and the `destination` is the root, then the pointer uses the entire path of the key. |

> | `parse_when` | No | String | Specifies under what conditions the processor should perform parsing. Default is no condition. Accepts a Data Prepper expression string following the [Data Prepper Expression Syntax]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/expression-syntax/). |

Reviewer: ```suggestion
| `parse_when` | No | String | Specifies under what conditions the processor should perform parsing. Default is no condition. Accepts a Data Prepper expression string following the [Data Prepper Expression syntax]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/expression-syntax/). |
```

> | `parse_when` | No | String | Specifies under what conditions the processor should perform parsing. Default is no condition. Accepts a Data Prepper expression string following the [Data Prepper Expression syntax]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/expression-syntax/). |

Reviewer: ```suggestion
| `parse_when` | No | String | Specifies under which conditions the processor should perform parsing. Default is no condition. Accepts a Data Prepper expression string following the [Data Prepper Expression syntax]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/expression-syntax/). |
```

> | `overwrite_if_destination_exists` | No | Boolean | If set to `true` then overwrite the destination. Set to `false` to prevent changing a destination value that exists. Defaults to `true`. |

Reviewer: ```suggestion
| `overwrite_if_destination_exists` | No | Boolean | Overwrites the destination if set to `true`. Set to `false` to prevent changing a destination value that exists. Defaults is `true`. |
```

> | `delete_source` | No | Boolean | If set to `true` then this will delete the source field. Defaults to `false`. |

Reviewer: ```suggestion
| `delete_source` | No | Boolean | If set to `true`, then the source field is deleted. Defaults is `false`. |
```

> | `delete_source` | No | Boolean | If set to `true` then overwrite the destination. Set to `false` to prevent changing a destination value that exists. Defaults to `true` |

Reviewer: @dlvenable @chenqi0805 The description for delete_source is the same as line 30. Is this intentional?

Author: That was a mistake on my part. I just pushed a new change that should correct this.
